app/src/main/java/com/ncnf/database/builder/UserBuilder.java

```
@@ -12,35 +9,47 @@
+
+    @Inject
+    DatabaseService db;
 
     @Override
     public User toObject(String uuid, Map<String, Object> data) {
 
-        String username = (String) data.get(USERNAME_KEY);
-        String email = (String) data.get(EMAIL_KEY);
-        String firstName = (String) data.get(FIRST_NAME_KEY);
-        String lastName = (String) data.get(LAST_NAME_KEY);
-        List<String> friends = (List<String>) data.get(FRIENDS_KEY);
-        List<String> ownedEvents = (List<String>) data.get(OWNED_EVENTS_KEY);
-        List<String> savedEvents = (List<String>) data.get(SAVED_EVENTS_KEY);
-        Log.d(DEBUG_TAG, data.get(BIRTH_DATE_KEY).toString());
-        Date birthDate = ((Timestamp) data.get(BIRTH_DATE_KEY)).toDate();
-        boolean notifications = (boolean) data.get(NOTIFICATIONS_KEY);
+        try {
+            String username = (String) data.get(USERNAME_KEY);
+            String email = (String) data.get(EMAIL_KEY);
+            String firstName = (String) data.get(FIRST_NAME_KEY);
+            String lastName = (String) data.get(LAST_NAME_KEY);
+            List<String> friends = (List<String>) data.get(FRIENDS_KEY);
+            List<String> ownedGroups = (List<String>) data.get(OWNED_GROUPS_KEY);
+            List<String> participatingGroups = (List<String>) data.get(PARTICIPATING_GROUPS_KEY);
+            List<String> savedEvents = (List<String>) data.get(SAVED_EVENTS_KEY);
+            Date birthDate = ((Timestamp) data.get(BIRTH_DATE_KEY)).toDate();
+            boolean notifications = (boolean) data.get(NOTIFICATIONS_KEY);
 
-        return new User(new DatabaseService(), uuid, username, email, firstName, lastName, friends, ownedEvents, savedEvents, birthDate, notifications);
+            return new User(db, uuid, username, email, firstName, lastName, friends, ownedGroups, participatingGroups, savedEvents, notifications, birthDate);
+        } catch (Exception e){
+            return null;
+        }
```

That's kind of brutal. In case of bug, it silently fails without insightful errors.

---

Yes it is a bit brutal but I don't really know another way to do it : the method could throw an exception but I don't know if that's a good solution.